src/GazeboUnderwater.cpp

```diff
@@ -20,82 +20,153 @@ namespace gazebo_underwater
+                        + " not found in gazebo world " + world->GetName();
+                gzthrow(msg);
+            }
         }else{
-            gzmsg << "GazeboUnderwater: model not found. Quit simulation... " << std::endl;
+            gzthrow("GazeboUnderwater: model_name not defined in world file !");
         }
 
-        size = getParameter<math::Vector3>("size","meters",
-                math::Vector3(1, 1, 1));
+        if(sdf->HasElement("link_name"))
+        {
+            link = model->GetLink( sdf->Get<std::string>("link_name") );
+            if( link ){
+                gzmsg << "GazeboUnderwater: link: " << link->GetName() << std::endl;
+                physics::InertialPtr modelInertia = link->GetInertial();
+                gzmsg << "GazeboUnderwater: link mass: " << modelInertia->GetMass() << std::endl;
+            }else{
+                std::string msg = "GazeboUnderwater: link " + sdf->Get<std::string>("link_name")
+                        + " not found in model " + model->GetName();
+                gzthrow(msg);
+            }
+        }else{
+            gzthrow("GazeboUnderwater: link_name not defined in world file !");
+        }
+
+        waterLevel = getParameter<double>("water_level","meters", 0.0);
+        fluidVelocity = getParameter<math::Vector3>("fluid_velocity","m/s",math::Vector3(0,0,0));
+        densityOfFluid = getParameter<double>("fluid_density","kg/m3", 1027);
+        dragCoefficient = getParameter<math::Vector3>("drag_coefficient","dimensionless",
+                math::Vector3(1,1,1));
+        // buoyancy must be the buoyancy when the model is completely submersed
+        buoyancy = getParameter<double>("buoyancy","N",0);
         centerOfBuoyancy = getParameter<math::Vector3>("center_of_buoyancy","meters",
-                math::Vector3(0, 0, 0.1));
-        fluidVelocity = getParameter<math::Vector3>("fluid_velocity","m/s",
-                math::Vector3(0,0,0));
-        viscousDamping = getParameter<math::Vector3>("viscous_damping","dimensionless",
-                math::Vector3(20,30,50));
-        waterLevel = getParameter<double>("water_level","meters", 2.0);
-        densityOfFluid = getParameter<double>("fluid_density","kg/m3", 1000);
+                math::Vector3(0, 0, 0.15));
+        volume = getParameter<double>("volume","meter3",1);
+
+        // If side_areas are not given in world file we use the bouding box dimensions to calculate it
+        if(sdf->HasElement("side_areas"))
+        {
+            sideAreas = getParameter<math::Vector3>("side_areas","meter2",
+                    math::Vector3(0.0,0.0,0.0));
+            if( sideAreas == math::Vector3(0.0,0.0,0.0) )
+                gzthrow("GazeboUnderwater: side_areas cannot be (0.0, 0.0, 0.0).");
+        }else{
+            math::Box linkBoudingBox = link->GetBoundingBox();
+            sideAreas = getParameter<math::Vector3>("side_areas","meter2",
```

Comment:
I would personally amend (i.e. rewrite the history). You don't need to rebase: since it is the last commit, you can amend it (there's an option for that in git gui)